src/ripple/app/tests/Offer.test.cpp

```diff
@@ -795,6 +796,64 @@ class Offer_test : public beast::unit_test::suite
             owners ("eve", 1));
     }
 
+    void
+    testRmMadeUnfunded()
+    {
+        // test cases for re-using offers found unfunded
+        testcase ("Made Unfunded");
+
+        // Create an offer that is made unfunded in the reverse pass, hits a
+        // limiting step, and confirms the made unfunded offer is not re-used
+        // when re-executing the limiting step.
+
+        using namespace jtx;
+        Env env (*this, features (featureFlowV2));
+
+        auto const alice = Account ("alice");
+        auto const bob = Account ("bob");
+        auto const carol = Account ("carol");
+        auto const gw = Account ("gw");
+        auto const USD = gw["USD"];
+        auto const EUR = gw["EUR"];
+
+        env.fund (XRP (10000), alice, bob, carol, gw);
+        env.trust (USD (10000), alice, bob, carol);
+        env.trust (EUR (10000), alice, bob, carol);
+
+        env (pay (gw, bob, USD (100)));
+        env (pay (gw, carol, USD (100)));
+        env (pay (gw, carol, EUR (100)));
+
+        // This will be made unfunded in the reverse pass
+        // Confirm that is isn't used in the forward pass
```

Comment:
This comment is really important.  Without it I would not have figured out why bob's offer of USD for XRP was removed from the book.